### PR TITLE
Default new developer clusters to etcd3 and protobuf serialization

### PR DESCRIPTION
**Pull request by smarterclayton**
There is a bug in etcd3 + Go 1.7 that still has to be resolved, but at least testing here

[test]

### CODE REVIEW

File: hack/install-etcd.sh [hunk 1]
```diff
@@ -1,6 +1,10 @@
 #!/bin/bash
 source "$(dirname "${BASH_SOURCE}")/lib/init.sh"
 
+set -o errexit
```
Comment (stevekuznetsov) [inline hack/install-etcd.sh]: This happens in `hack/lib/init.sh` and is not necessary anywhere else.

File: test/cmd/basicresources.sh [hunk 2]
```diff
@@ -26,7 +26,8 @@ os::test::junit::declare_suite_start "cmd/basicresources/versionreporting"
 os::build::get_version_vars
 os_git_regex="$( escape_regex "${OS_GIT_VERSION%%-*}" )"
 kube_git_regex="$( escape_regex "${KUBE_GIT_VERSION%%-*}" )"
-etcd_git_regex="$( escape_regex "${ETCD_GIT_VERSION%%-*}" )"
+etcd_version="$(echo "${ETCD_GIT_VERSION}" | sed -E "s/\-.*//g" | sed -E "s/v//")"
```
Comment (stevekuznetsov) [inline test/cmd/basicresources.sh]: Can we achieve this transform without hierloglyphics? Nobody wants to see a string of `seds`

File: hack/install-etcd.sh [hunk 3]
```diff
@@ -23,7 +27,7 @@ fi
 
 # setup a private GOPATH so the build can succeed
 export GOPATH="${PWD}/gopath"
-rm -f "${GOPATH}/src/github.com/coreos/etcd"
+rm -rf "${GOPATH}/src/github.com/coreos/etcd"
```
Comment (stevekuznetsov) [inline hack/install-etcd.sh]: Be careful: https://github.com/openshift/origin/commit/0c804382069f5592727a853015729d34a9f5bf8c

**Comment by smarterclayton** [comment]
On the sed thing I really don't want to rewrite this use case and the
version logic in this PR.

**Comment by smarterclayton** [comment]
@stevekuznetsov see #10976 which is the updated version with your comment applied

**Comment by stevekuznetsov** [comment]
@smarterclayton ack

**Comment by smarterclayton** [comment]
@timothysc

**Comment by smarterclayton** [comment]
Blocked on https://github.com/coreos/etcd/issues/6565 (until we figure out why client certs don't work)

**Comment by smarterclayton** [comment]
@timothysc updated to etcd v3.1.0-rc.0 which fixes the client cert change, hopefully this will result in passes

**Comment by smarterclayton** [comment]
[test]

**Comment by timothysc** [comment]
Has anyone verified that you the rev'd client can indeed talk with a 3.0.12+ server?  

-alpha.1 could not, https://bugzilla.redhat.com/show_bug.cgi?id=1387391

**Comment by smarterclayton** [comment]
@mfojtik @kargakis the deployment flakes here are either because the deployment controller is going crazy, or is just plain faster now, which means that these were always latent flakes.  Can you look on monday?

**Comment by kargakis** [comment]
> @mfojtik @kargakis the deployment flakes here are either because the deployment controller is going crazy, or is just plain faster now, which means that these were always latent flakes. Can you look on monday?

Tested and https://github.com/openshift/origin/pull/11412 seems to restore things back to normal. I also have some code locally that reduces the competition between the trigger and the main controller.

**Comment by kargakis** [comment]
[test]

**Comment by brenton** [comment]
@timothysc, Am I correct in thinking this blocks most (or all) of your etcd3 testing?  I want to make sure we get the correct labels on this PR so it's brought up on any blocker calls.

**Comment by timothysc** [comment]
> Am I correct in thinking this blocks most (or all) of your etcd3 testing?

@brenton correct.

**Comment by brenton** [comment]
@smarterclayton, if I didn't apply the correct label please change it.  Should this be assigned to you or someone else?

**Comment by smarterclayton** [comment]
me.

**Comment by smarterclayton** [comment]
[test]

**Comment by smarterclayton** [comment]
[merge] if no other comments.

This is a bump to etcd 3.1.0-rc.0.  We would pick the release etcd.  This is only for all in one servers, and development workloads.  All other paths will continue to work as normal.

@sdodson in the 1.4 Ansible code, if you use --write-config with the embedded etcd you will be set to protobuf.  My understanding is that no ansible code path today uses embedded etcd, and if it does, it does not use --write config.  Can you confirm?

**Comment by sdodson** [comment]
> @sdodson in the 1.4 Ansible code, if you use --write-config with the embedded etcd you will be set to protobuf. My understanding is that no ansible code path today uses embedded etcd, and if it does, it does not use --write config. Can you confirm?

Today that's incorrect, if you have a single master install and you've not defined etcd hosts you will get embedded etcd today. We're hoping to make sure that's not the case for 1.4/3.4.

**Comment by smarterclayton** [comment]
Hoping, or going to?

**Comment by smarterclayton** [comment]
@kargakis potential race in deployment config exposed by much faster
apiserveres.

On Oct 20, 2016, at 8:13 PM, OpenShift Bot <notifications@github.com> wrote:

continuous-integration/openshift-jenkins/test FAILURE (
https://ci.openshift.redhat.com/jenkins/job/test_pr_origin/10355/) (Base
Commit: e1d9196
<https://github.com/openshift/origin/commit/e1d9196e449950f403bf8f93d4608dd92dcbb845>
)

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/10980#issuecomment-255261903>,
or mute
the thread

**Comment by kargakis** [comment]
> @kargakis potential race in deployment config exposed by much faster apiserveres.

Got any logs from the master? What failures did you see? Update conflicts or something else?

**Comment by smarterclayton** [comment]
It was update conflicts in many different deployment related tests.

**Comment by kargakis** [comment]
https://github.com/openshift/origin/pull/11412 should reduce conflicts but
since we have added a bunch of fields in the status in 1.3 and now
Conditions in 1.4, it's inherent that we are going to have more update
conflicts. We need to switch a couple of our commands to use PATCH (delete,
rollout pause/resume, rollout cancel once we have it) and I think `oc
rollout latest` should use GuaranteedUpdate similar to how the upstream
Rollback rest already uses it:
https://github.com/kubernetes/kubernetes/blob/5fc18ab1bcdaf107509c29960e8081ae15950ec7/pkg/registry/extensions/deployment/etcd/etcd.go#L176

**Pull request merged**